src/hooks/useLocations.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+
+export const useLocations = (lat, lng) => {
+  const [locations, setLocations] = useState([]);
+  const [status, setStatus] = useState('idle');
+  const [error, setError] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+
+  useEffect(() => {
+    // Abort fetch
+    let abortController = new AbortController();
+    const { signal } = abortController;
+    signal.addEventListener('abort', () => {
+      // Logs true:
+      console.log(signal.aborted);
+    });
+
+    // Call to API
+    const baseURL = 'https://segdeha.com/api/nearby.php';
+    const completeURL = lat && lng && `${baseURL}?lat=${lat}&lng=${lng}`;
+
+    setStatus('loading');
+    fetch(completeURL, { signal })
+      .then((response) => {
+        setIsFetching(true);
+        return response.json();
+      })
+      .then((data) => {
+        setLocations(data?.query?.pages);
+        setStatus('success');
+      })
+      .catch((err) => {
+        setStatus('error');
+        setError(err);
+        if (err.name === 'AbortError') {
```

Comment:
Oh, also, I wonder if we want to keep showing the map, when dragging and fetching new locations...question for the group, I think.

Reply:
We’ve been using async functions in the useFetch.js. It would great to  keep doing that across the whole project 😄

Reply:
We’ve been using ``async`` functions in the useFetch.js. It would great to  keep doing that across the whole project😄

Reply:
@laurenmbeatty : I'm so sorry I forgot to remove it. I put this check because I wanted to identify if our fetch was canceled, and yes, it works. I will remove this code because we don't need it more.

Reply:
@katia-utochkina : We have changed how we can get data from API, also created another custom hook: `useLocations`. Thanks for your suggestion 😄!